src/auxiliary/mod.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module is about interaction with auxiliary block chain.
+
+use futures::Future;
+use rlp;
+use sha3::{Digest, Keccak256};
+use web3::contract::Options;
+use web3::types::{Address, H160, H256};
+
+use super::ethereum::Ethereum;
+use ethereum::types::{Block, Error, ErrorKind};
+
+const REPORT_BLOCK_ESTIMATED_GAS: i32 = 3_000_000;
+
+/// Reports block on block store if not already reported.
+///
+/// # Arguments
+///
+/// * `block_chain` - A blockchain object.
+/// * `event_loop` - The reactor's event loop to handle the tasks spawned.
+/// * `block_store_address` - The address of block store.
+/// * `validator_address` - The address of validator address.
+pub fn report_block(
+    block_chain: &Ethereum,
+    event_loop: &tokio_core::reactor::Handle,
+    block_store_address: Address,
+    validator_address: Address,
+    block: &Block,
+) {
+    info!("Reporting block for number {:?} ", block.number);
+
+    let encoded_block = rlp::encode(block);
+    let block_hash = H256::from(Keccak256::digest(encoded_block.as_slice()).as_slice());
```

Comment:
Should we have a helper method somewhere that does these two steps combined as one function call? I could imagine it could be re-used. Not sure, though 🤔

Reply:
Do you suggest a separate `helper` module for this?